lib/workers/repository/update/pr/code-owners.spec.ts

```diff
@@ -22,13 +23,183 @@ describe('workers/repository/update/pr/code-owners', () => {
+          'packages/d/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package A and B to @maria (a), @jimmy (b), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/a/package.json',
+          'packages/b/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package A, B and C to @john, @maria (a), @jimmy (b), @dan (c), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/a/package.json',
+          'packages/b/package.json',
+          'packages/c/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@maria', '@jimmy', '@dan', '@john']);
+      });
+
+      it('assigns changes in package C and D to @dan (c), @maria (d), @jimmy (e), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/c/package.json',
+          'packages/d/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@dan', '@maria', '@jimmy', '@john']);
+      });
+
+      it('assigns changes in package D and E to @jimmy (d, e), @maria (d), @john (*)', async () => {
+        fs.readLocalFile.mockResolvedValueOnce(mockCodeOwners);
+        git.getBranchFiles.mockResolvedValueOnce([
+          'packages/d/package.json',
+          'packages/e/package.json',
+          'yarn.lock',
+        ]);
+        const codeOwners = await codeOwnersForPr(pr);
+        expect(codeOwners).toEqual(['@jimmy', '@maria', '@john']);
+      });
+    });
+
+    it('does not require all files to match a single rule, regression test for #12611', async () => {
+      fs.readLocalFile.mockResolvedValueOnce(
+        [
+          '* @reviewer-1 @reviewer-2 @reviewer-3 @reviewer-4 @reviewer-5',
+          'server/pom.xml @reviewer-1',
+          'client/package.json @reviewer-1',
+          'client/package-lock.json @reviewer-1',
+        ].join('\n')
+      );
+      git.getBranchFiles.mockResolvedValueOnce(['server/pom.xml']);
+      const codeOwners = await codeOwnersForPr(pr);
+      expect(codeOwners).toEqual([
+        '@reviewer-1', // matched by file
+        '@reviewer-2', // matched by wildcard
+        '@reviewer-3',
+        '@reviewer-4',
+        '@reviewer-5',
+      ]);
+
+      fs.readLocalFile.mockResolvedValueOnce(
+        [
```

Comment:
same

Reply:
Solved in c2888e3faa6c3ff7a3c74f5ea227b8ae4e19a92b